acceptance/tests/task_local.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'bolt_command_helper'
+# TODO I don't think we want to include the bolt gem here...
+require_relative '../../lib/bolt/util.rb'
+
+test_name "bolt tasks should run on the local transport" do
+  extend Acceptance::BoltCommandHelper
+
+  # From spec/bolt/transport/shared_examples
+  def posix_context
+    {
+      tmpdir: '/tmp',
+      mkdir: 'mkdir -p',
+      supported_req: 'shell',
+      extension: '.sh',
+      env_task: "#!/bin/sh\nprintenv PT_message_one\nprintenv PT_message_two",
+      stdin_task: "#!/bin/sh\ncat"
+    }
+  end
+
+  def windows_context
+    {
+      tmpdir: 'C:/mytmp',
+      mkdir: 'mkdir',
+      supported_req: 'powershell',
+      extension: '.ps1',
+      env_task: "Write-Output \"${env:PT_message_one}\n${env:PT_message_two}\"",
+      stdin_task: "$line = [Console]::In.ReadLine()\nWrite-Output \"$line\""
+    }
+  end
+
+  os_context = if Bolt::Util.windows?
+                 windows_context
+               else
+                 posix_context
+               end
+
+  dir = bolt.tmpdir('locomoco')
+  modulepath = File.join(dir, 'modules', 'test')
+  user = 'nopass'
+
+  step 'with task that reads env vars' do
+    step 'create task on bolt controller' do
+      on(bolt, "#{os_context[:mkdir]} #{File.join(modulepath, 'tasks')}")
+      create_remote_file(bolt, File.join(modulepath, 'tasks', "echo#{os_context['extension']}"), os_context[:env_task])
+    end
+
+    step 'execute `bolt task run` locally' do
+      bolt_command = "bolt task run test::echo message_one=beep message_two=boop"
+      flags = {
+        '--nodes' => 'localhost',
+        '--modulepath' => "#{dir}/modules"
+      }
+
+      result = bolt_command_on(bolt, bolt_command, flags)
+      assert_match(/Successful on 1 node: localhost/, result.stdout, "Unexpected output from the command:\n#{result.cmd}")
+      assert_match(/beep\n  boop/, result.stdout, "Unexpected output from the command:\n#{result.cmd}")
+      assert(result.exit_code == 0, "#{result.cmd} exited #{result.exit_code} with stderr: #{result.stderr}")
+    end
+  end
+
+  unless Bolt::Util.windows?
+    step 'as a user without a password' do
+      step 'create user who can escalate without password' do
+        on(bolt, "/opt/puppetlabs/bolt/bin/puppet resource user #{user} ensure=present")
```

Comment:
I would just use beaker for this. `apply_manifest_on` maybe.

Reply:
https://jenkins-platform.delivery.puppetlabs.net/job/experimental_auto_bolt_vanagon-integration-component_maint-no-password-bug_project-master/RMM_COMPONENT_TO_TEST_NAME=bolt,SLAVE_LABEL=beaker,TEST_TARGET=osx1013-64bolt%252Cssh.-centos7-64ssh.-windows2008-64winrm./1/testReport/junit/(root)/tests/task_local_rb/